### DISCUSSION TITLE
How to get influent concentration for a link?

### DISCUSSION THREAD

**Discussion by nssimmo2** [pyswmm/pyswmm#359] [Q&A]
Hello,

I am trying to retrieve the influent concentration for a link using getters and setters.  I'm looking through "toolkitapi.py" under the "LinkPollut" class and see a few options.  "reactorQual" gives the current concentration in the mixed reactor for the link, but not sure if "linkQual" or "totalLoad" have any relation to the inflow concentration/loading from the upstream node or where else I may need to look?

Ultimately, I am trying to counteract the "QCin" mass balance term for a link so that it doesn't receive any loading from upstream, but unsure how to reference the inflow concentration from the upstream node.  I suppose I could reference "NodePollut" and "reactorQual" to get this data but didn't know if there was a more direct way to do this by referencing only the link in question and not separately refencing the specific upstream node.

Thanks for any insight or direction on which parameter I may need to reference to get this information!

**Comment by abhiramm7** [comment]
Hi @nssimmo2, 

As you said, the concentration in the node upstream of a link to estimate inflow concentration is the best way to get at what you are looking for. We don't yet have a way to get Link Cin. I would get the pollutant from NodePollut. 

I hope this helps. Feel free to reach out if you have any questions. 

Cheers, 
Abhi
